src/api/liskApi.js

```diff
@@ -12,183 +12,177 @@
+	}
 
-		if (providedNethash) {
-			NetHash.nethash = providedNethash;
-			NetHash.version = '0.0.0a';
+	get randomNode() {
+		const nodes = this.nodes.filter(node => !this.isBanned(node));
+
+		if (!nodes.length) {
+			throw new Error(
+				'Cannot get random node: all relevant nodes have been banned.',
+			);
 		}
 
-		return NetHash;
+		const randomIndex = Math.floor(Math.random() * nodes.length);
+		return nodes[randomIndex];
 	}
 
-	/**
-	 * @method getNodes
-	 * @return {Object}
-	 */
-
-	getNodes() {
-		return {
-			official: this.defaultNodes.map(node => ({ node })),
-			ssl: this.defaultSSLNodes.map(node => ({ node, ssl: true })),
-			testnet: this.defaultTestnetNodes.map(node => ({
-				node,
-				testnet: true,
-			})),
-		};
+	get ssl() {
+		return this[ssl];
 	}
 
-	/**
-	 * @method setNode
-	 * @param {String} node
-	 * @return {Object}
-	 */
-
-	setNode(node) {
-		this.node = node || privateApi.selectNewNode.call(this);
-		return this.node;
-	}
+	set ssl(newSSLValue) {
+		if (this.ssl !== newSSLValue) {
+			const nonSSLPort = this.testnet ? TEST_PORT : LIVE_PORT;
 
-	/**
-	 * @method setTestnet
-	 * @param {Boolean} testnet
-	 */
+			this[ssl] = newSSLValue;
+			this.port = newSSLValue ? SSL_PORT : nonSSLPort;
+			this[bannedNodes] = [];
 
-	setTestnet(testnet) {
-		if (this.testnet !== testnet) {
-			this.bannedNodes = [];
+			this.selectNewNode();
 		}
-		this.testnet = testnet;
-		this.port = testnet ? TEST_PORT : LIVE_PORT;
+	}
 
-		privateApi.selectNewNode.call(this);
+	get testnet() {
+		return this[testnet];
 	}
 
-	/**
-	 * @method setSSL
-	 * @param {Boolean} ssl
-	 */
+	set testnet(newTestnetValue) {
+		if (this.testnet !== newTestnetValue) {
+			const nonTestnetPort = this.ssl ? SSL_PORT : LIVE_PORT;
+
+			this[testnet] = newTestnetValue;
+			this.port = newTestnetValue ? TEST_PORT : nonTestnetPort;
+			this[bannedNodes] = [];
 
-	setSSL(ssl) {
-		if (this.ssl !== ssl) {
-			this.ssl = ssl;
-			this.bannedNodes = [];
-			privateApi.selectNewNode.call(this);
+			this.selectNewNode();
 		}
 	}
 
-	/**
-	 * @method broadcastSignedTransaction
-	 * @param transaction
-	 * @param callback
-	 *
-	 * @return {Object}
-	 */
+	banActiveNode() {
+		if (!this.isBanned(this.node)) {
+			this[bannedNodes].push(this.node);
+			this.node = this.selectNewNode();
+			return true;
+		}
+		return false;
+	}
 
-	broadcastSignedTransaction(transaction, callback) {
+	broadcastSignatures(signatures, callback) {
 		const request = {
-			requestUrl: `${utils.getFullURL(this)}/api/transactions`,
+			requestUrl: `${utils.getFullURL(this)}/api/signatures`,
 			nethash: this.nethash,
-			requestParams: { transaction },
+			requestParams: { signatures },
 		};
-
 		return privateApi.sendRequestPromise
 			.call(this, POST, request)
 			.then(result => result.body)
 			.then(utils.optionallyCallCallback.bind(null, callback));
 	}
 
-	broadcastSignatures(signatures, callback) {
+	broadcastSignedTransaction(transaction, callback) {
```

Comment:
Are we leaving callback for compatibility?

Reply:
No, this PR was made before https://github.com/LiskHQ/lisk-js/pull/545 was merged and will need to be updated.